### ISSUE TITLE
One-var incorrectly flags declarations

### ISSUE THREAD

**Issue by ilyavolodin** [eslint/eslint#5363]
**What version of ESLint are you using?**
v2.2.0
**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**
```js
{
  "one-var": 2
}
```
**What did you do? Please include the actual source code causing the issue.**
```js
var fs = require('fs'),
    path = require('path');

try {
    var result = fs.readFileSync(path.resolve(file));
} catch (e) {
    throw new Error('Couldn\'t read file', file, e);
}
```
**What did you expect to happen?**
No error
**What actually happened? Please include the actual, raw output from ESLint.**
```sh
5:5 - Combine this with the previous 'var' statement. (one-var)
```

Don't think this is correct behavior, since there's a `try` statement between variable declarations.

**Comment by alberto** [comment]
I think the rule is working as intended since there is no block scope.

**Comment by kaicataldo** [comment]
I think what @alberto said is correct. `catch()` is a function, but the `try` block is not, so it is in the same scope as the first two `var`s. I tried this in the Chrome's console and this was the result:
```
try {
  var test = 0;
} catch(e) {
  var test2 = 1;
}

console.log(test);
console.log(test2);
```
Executing the above code yields:
```
0
undefined
```

**Comment by platinumazure** [comment]
@ilyavolodin This seems like correct behavior to me as there is no new scope. I think the "correct code" in that case is as follows:

```js
var fs = require('fs'),
    path = require('path'),
    result;

try {
    result = fs.readFileSync(path.resolve(file));
} catch (e) {
    throw new Error('Couldn\'t read file', file, e);
}
```

**Comment by diorahman** [comment]
How about using `let`? 

Does `try {` open a new scope?

**Comment by ilyavolodin** [comment]
Now that I had time to think about it, it does look like the rule is doing the right thing. It's a bit inconvenient from time to time, but it just wants all of the declarations under one `var` statement, which makes sense.
@diorahman Not sure if `try` is treated as a block for `let`. You might want to try it with `one-var` rule, and if it behaves incorrectly, open a new issue.
